tests/modules/images/webp-uploads/webp-uploads-test.php

```diff
@@ -280,31 +257,18 @@ public function it_should_create_a_webp_version_with_all_the_required_properties
 		$file    = get_attached_file( $attachment_id, true );
 		$dirname = pathinfo( $file, PATHINFO_DIRNAME );
 
-		$this->assertArrayHasKey( 'image/jpeg', $metadata['sources'] );
-		$this->assertIsArray( $metadata['sources']['image/jpeg'] );
-		$this->assertArrayHasKey( 'file', $metadata['sources']['image/jpeg'] );
-		$this->assertArrayHasKey( 'filesize', $metadata['sources']['image/jpeg'] );
+		$this->assertImageHasSource( 'image/jpeg', $attachment_id );
 		$this->assertStringEndsWith( $metadata['sources']['image/jpeg']['file'], $file );
 		$this->assertFileExists( path_join( $dirname, $metadata['sources']['image/jpeg']['file'] ) );
 		$this->assertSame( $metadata['sources']['image/jpeg']['filesize'], filesize( path_join( $dirname, $metadata['sources']['image/jpeg']['file'] ) ) );
 
-		$this->assertArrayHasKey( 'image/webp', $metadata['sources'] );
-		$this->assertIsArray( $metadata['sources']['image/webp'] );
-		$this->assertArrayHasKey( 'file', $metadata['sources']['image/webp'] );
-		$this->assertArrayHasKey( 'filesize', $metadata['sources']['image/webp'] );
+		$this->assertImageHasSource( 'image/webp', $attachment_id );
 		$this->assertStringEndsWith( '.webp', $metadata['sources']['image/webp']['file'] );
 		$this->assertFileExists( path_join( $dirname, $metadata['sources']['image/webp']['file'] ) );
 		$this->assertSame( $metadata['sources']['image/webp']['filesize'], filesize( path_join( $dirname, $metadata['sources']['image/webp']['file'] ) ) );
 
-		$this->assertArrayHasKey( 'sources', $metadata['sizes']['thumbnail'] );
-		$this->assertArrayHasKey( 'image/jpeg', $metadata['sizes']['thumbnail']['sources'] );
-		$this->assertArrayHasKey( 'filesize', $metadata['sizes']['thumbnail']['sources']['image/jpeg'] );
-		$this->assertArrayHasKey( 'file', $metadata['sizes']['thumbnail']['sources']['image/jpeg'] );
-		$this->assertArrayHasKey( 'image/webp', $metadata['sizes']['thumbnail']['sources'] );
-		$this->assertArrayHasKey( 'filesize', $metadata['sizes']['thumbnail']['sources']['image/webp'] );
-		$this->assertArrayHasKey( 'file', $metadata['sizes']['thumbnail']['sources']['image/webp'] );
-		$this->assertStringEndsNotWith( '.jpeg', $metadata['sizes']['thumbnail']['sources']['image/webp']['file'] );
-		$this->assertStringEndsWith( '.webp', $metadata['sizes']['thumbnail']['sources']['image/webp']['file'] );
+		$this->assertImageHasSizeSource( 'image/jpeg', 'thumbnail', $attachment_id );
+		$this->assertImageHasSizeSource( 'image/webp', 'thumbnail', $attachment_id );
```

Comment:
I think the abstraction works great to ensure the presence of the key this 2 assertions would ensure that the correct file extensions are used and can be used after `assertImageHasSizeSource` as at that point if the previous assertions pass it can safely be used at this stage.

```suggestion
		$this->assertImageHasSizeSource( 'image/webp', 'thumbnail', $attachment_id );
		$this->assertStringEndsNotWith( '.jpeg', $metadata['sizes']['thumbnail']['sources']['image/webp']['file'] );
		$this->assertStringEndsWith( '.webp', $metadata['sizes']['thumbnail']['sources']['image/webp']['file'] );		
```

Reply:
I have added extensions check to image source constraints.